projects/pt1/python/torch_mlir_e2e_test/test_suite/elementwise.py

```diff
@@ -6845,3 +6845,52 @@ def forward(self):
 @register_test_case(module_factory=lambda: TrilIndicesOfssetGreaterThanRowModule())
 def TrilIndicesOfssetGreaterThanRowModule_basic(module, tu: TestUtils):
     module.forward()
+
+
+# ==============================================================================
+
+
+class GCDModule(torch.nn.Module):
+    @export
+    @annotate_args([None, [(4, 4), torch.int32, True], [(4, 4), torch.int32, True]])
+    def forward(self, A, B):
+        return torch.gcd(A, B)
+
+
+@register_test_case(module_factory=lambda: GCDModule())
+def GCDModule_I32(module, tu: TestUtils):
+    A = tu.rand(4, 4).to(dtype=torch.int32)
+    B = tu.rand(4, 4).to(dtype=torch.int32)
```

Comment:
I think this only checks elements in the range 0-1 because the defaults to high and low are 0 and 1. I was looking at this before the force push and changing it to [0, 100] or [-100, 100] were causing test failures (but looks like it works now).

Reply:
Okay, I have changed the tests to generate values between -100 and 100. 
Also yes, in first version there was an error so it didn't calculate the correct results, it's fixed now.